### DISCUSSION TITLE
Unable to add SPICE data to Galileo Cube Files

### DISCUSSION THREAD

**Discussion by bensonyee88** [DOI-USGS/ISIS3#5494] [Q&A]
Hello! We're facing issues with adding SPICE data to our Galileo cube files using the spiceinit function. I have pasted the error message below. Would really appreciate any help!

Last login: Sun May  5 22:53:24 on ttys000
(base) benson@dhcp-vl2041-18708 ~ % conda activate isis
(isis) benson@dhcp-vl2041-18708 ~ % spiceinit
(isis) benson@dhcp-vl2041-18708 ~ % qimage 
zsh: command not found: qimage
(isis) benson@dhcp-vl2041-18708 ~ % ls *.cub
7401r.cub	7452r.cub	8700r.cub	8726r.cub
(isis) benson@dhcp-vl2041-18708 ~ % qview 8700r.cub
^C
(isis) benson@dhcp-vl2041-18708 ~ % spiceinit from=8700r.cub attach=yes
**ERROR** No existing files found with a numerical version matching [kernels.????.db] in [/Applications/anaconda3/envs/isis/data/base/kernels/lsk].
(isis) benson@dhcp-vl2041-18708 ~ % spiceinit from=8700r.cub web=yes   
**ERROR** The server sent an unrecognized response.
(isis) benson@dhcp-vl2041-18708 ~ % ls *.cub
7401r.cub	7452r.cub	8700r.cub	8726r.cub
(isis) benson@dhcp-vl2041-18708 ~ % spiceinit from=8726r.cub web=yes
**ERROR** The server sent an unrecognized response.
(isis) benson@dhcp-vl2041-18708 ~ % gllssi2isis from=8700r.lbl to=8700r_v2.cub
(isis) benson@dhcp-vl2041-18708 ~ % spiceinit from=8700r_v2.cub web=true
**ERROR** The server sent an unrecognized response.
(isis) benson@dhcp-vl2041-18708 ~ % spiceinit from=8700r_v2.cub attach=true
**ERROR** No existing files found with a numerical version matching [kernels.????.db] in [/Applications/anaconda3/envs/isis/data/base/kernels/lsk].
(isis) benson@dhcp-vl2041-18708 ~ %

**Comment by amystamile-usgs** [comment]
Seems like there may be a bug with our web spice server for this particular image. Please post an [issue](https://github.com/DOI-USGS/ISIS3/issues) so that we can look into what is causing the "The server sent an unrecognized response" error. In the meantime, you can install the isis data directly. See instructions here: https://github.com/DOI-USGS/ISIS3?tab=readme-ov-file#the-isis-data-area.

**Reply by bensonyee88** [reply]
Thank you for your reply! We have downloaded isis data directly using "downloadIsisData galileo $ISISDATA" and attached the cube files. This is the error message we got: 

*ERROR* No existing files found with a numerical version matching [kernels.????.db] in [/Applications/anaconda3/envs/isis/data/base/kernels/lsk]

**Reply by acpaquette** [reply]
@bensonyee88 I think you will also need to download the base area, so along with your command for downloading the galileo area you will also need the following:

`downloadIsisData base $ISISDATA`

**Reply by bensonyee88** [reply]
Oh it works now, thank you!! We have added the spice data and radiometrically calibrated the cube file. Do you happen to know of any way to generate a flat DEM for the sfs function? We are trying to create a DEM for our cube file.

**Reply by jlaura** [reply]
@bensonyee88 Here are few questions and thoughts:

- Are you confident that you have an adequate number of observations with similar resolutions and varying illuminations (E/W/nadir at least) to be able to apply SfS?
- If so, you are going to want to generate a low resolution DTM to get started. The [SfS docs](https://stereopipeline.readthedocs.io/en/latest/sfs_usage.html#) in the NASA Ames Stereo Pipeline are a great start. As is [Oleg's paper](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2018EA000390). If you have seen both of these before, apologies for being pedantic!
- If these are Europa data, have you seen Mike Bland's [released DTMs](https://stac.astrogeology.usgs.gov/browser-dev/#/api/collections/galileo_usgs_photogrammetrically_controlled_dtms)? Are those helpful?

**Reply by lwellerastro** [reply]
Hi @bensonyee88, there are also smithed camera pointing for most Galileo SSI images of Europa. You can pick those up via spiceinit and setting the parameter cksmithed=true. Look for the following keywords in the Kernels group of the cube label to confirm (may also print to screen during spiceinit or be captured in your print.prt depending on you IsisPreferences settings):
```
    InstrumentPositionQuality = Reconstructed
    InstrumentPointingQuality = Smithed
```
There are no smithed spacecraft position kernels for these images (InstrumentPositionQuality), but improved camera pointing (InstrumentPointingQuality) has been applied. This will be essential for DTM generation as many of these data have very poor default spice.
 
The particular image you include in your post has smithed pointing information.

Reference to the control work for Europa:https://agupubs.onlinelibrary.wiley.com/doi/10.1029/2021EA001935
